src/common/casualReferences.ts

```diff
@@ -52,3 +70,53 @@ export function tonight(reference: ReferenceWithTimezone, implyHour = 22): Parsi
     assignSimilarDate(component, targetDate);
     return component;
 }
+
+export function lastNight(reference: ReferenceWithTimezone, implyHour = 0): ParsingComponents {
+    let targetDate = dayjs(reference.instant);
+    const component = new ParsingComponents(reference, {});
+    if (targetDate.hour() < 6) {
+        targetDate = targetDate.add(-1, "day");
+    }
+    assignSimilarDate(component, targetDate);
+    component.imply("hour", implyHour);
+    return component;
+}
+
+export function evening(reference: ReferenceWithTimezone, implyHour = 20): ParsingComponents {
```

Comment:
This function could be replaced with `tonight(ref, 20)`, right?

Reply:
Actually I'm not sure about this. Because `tonight` has known date and `evening` doesn't. It's more like `morning`, which is just morning with unknown date.